### PR TITLE
Get Full Kubernetai Zone Lists

### PR DESCRIPTION
**Pull request by Rigdon**
This changes the AutoPath func to use the zone list for each stanza to build the search path.

### CODE REVIEW

File: plugin/kubernetai/kubernetai.go [hunk 1]
```diff
@@ -83,14 +87,25 @@ func (k8i Kubernetai) ServeDNS(ctx context.Context, w dns.ResponseWriter, r *dns
...
 // AutoPath routes AutoPath requests to the authoritative kubernetes.
 func (k8i Kubernetai) AutoPath(state request.Request) []string {
+	var searchPath []string
 	for _, k := range k8i.Kubernetes {
-		zone := plugin.Zones(k.Zones).Matches(state.Name())
-		if zone == "" {
-			continue
+		zones := make([]string, 0, len(k.Zones)*2)
+		zones = append(zones, k.Zones...)
+		for _, z := range k.Zones {
```
Comment (chrisohaver) [inline plugin/kubernetai/kubernetai.go]: This section isn't quite right.  It needs to build the full search list for each unique zone...  so for 2 unique zones `cluster1.local.` and `cluster2.local.` we should generate...

~~~
<client-pod-namespace>.svc.cluster1.local.
svc.cluster1.local.
cluster1.local.
<client-pod-namespace>.svc.cluster2.local.
svc.cluster2.local.
cluster2.local.
<cluster-search-domains>
~~~

<client-pod-namespace> = the namespace of the client pod
<cluster-search-domains> = the search domains on the k8s host nodes. 

or breadth-first ...

~~~
<client-pod-namespace>.svc.cluster1.local.
<client-pod-namespace>.svc.cluster2.local.
svc.cluster1.local.
svc.cluster2.local.
cluster1.local.
cluster2.local.
<cluster-search-domains>
~~~

**Comment by chrisohaver** [comment]
@Rigdon, have you had a chance to come back to this PR?

**Comment by chrisohaver** [comment]
:( the tests were just flaking before... but as of about 30 minutes ago they are totally hosed.  Looking into it...  Never had so many different problems at the same time with this.  I'm guessing someone did some changes/updates on the test server.

**Comment by chrisohaver** [comment]
OK - looks as if some one added overly aggressive "log rotation" to the CI log output, which was deleting the log viewer.html.  I restored it for now ... need to track down what is deleting directory.

File: plugin/kubernetai/kubernetai.go [hunk 2]
```diff
@@ -83,14 +88,32 @@ func (k8i Kubernetai) ServeDNS(ctx context.Context, w dns.ResponseWriter, r *dns
...
+			}
+		}
+		zone := plugin.Zones(zones).Matches(state.Name())
+		if zone != "" {
+			searchPath = append([]string{zone}, searchPath...)
+			ip := state.IP()
+			pods := k.APIConn.PodIndex(ip)
+			var pod *api.Pod = nil
+			if len(pods) != 0 {
+				pod := ps[0]
```
Comment (chrisohaver) [inline plugin/kubernetai/kubernetai.go]:
~~~
plugin/kubernetai/kubernetai.go:107:12: undefined: ps
Makefile:11: recipe for target 'coredns' failed
~~~

**Comment by Rigdon** [comment]
@chrisohaver are these tests still flaky or these real failures now?

**Comment by chrisohaver** [comment]
Im not sure.  When you test these cases by hand, to they work?

**Comment by chrisohaver** [comment]
Would be good to add some unit tests for AutoPath, until I get the integration tests mystery solved...

**Pull request closed without merging**